src/lrtd.rs

```diff
@@ -0,0 +1,256 @@
+//! Utility to help with "really nice to add a warning for tasks that might be blocking"
+use libc;
+use rand::thread_rng;
+use rand::Rng;
+use std::collections::HashSet;
+use std::sync::mpsc;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use std::{env, thread};
+use tokio::runtime::{Builder, Runtime};
+
+const PANIC_WORKER_BLOCK_DURATION_DEFAULT: Duration = Duration::from_secs(60);
+
+fn get_panic_worker_block_duration() -> Duration {
+    let duration_str = env::var("MY_DURATION_ENV").unwrap_or_else(|_| "60".to_string());
+    duration_str
+        .parse::<u64>()
+        .map(Duration::from_secs)
+        .unwrap_or(PANIC_WORKER_BLOCK_DURATION_DEFAULT)
+}
+
+fn get_thread_id() -> libc::pthread_t {
+    unsafe { libc::pthread_self() }
+}
```

Comment:
is there a reason this code uses `libc::pthread_t` instead of Rust's `std::thread::ThreadId`? it might be nicer to use Rust `ThreadId`s so that user code can more easily correlate the blocking threads passed to `BlockingActionHandler` with Rust thread IDs, since they're not guaranteed to be the same as `libc::pthread_t`s?

Reply:
Oh there is, I actually tried to use std::thread::ThreadId... but I was not able to find a way to convert it to a libc::pthread_t.... which is needed for libc::pthread_kill  (see test_blocking_detection_multi_capture_stack_traces for more context)

Reply:
ah, yeah, i see the use of `pthread_kill` for capturing stack traces, hmm. i think it's unfortunate for the long-running task detector module to expose `libc` types in its API surface, because it would be nice for this module to not require `cfg(unix)`...ideally, the downstream implementations of the trait could be OS-specific, instead. however, i dunno if there's an easy way to do this; i suppose an implementation of the trait for Unix platforms could internally store a hash map or something that associates Rust `ThreadId`s with `libc::pthread_t`s, but that introduces additional overhead... i dunno.

Reply:
> i suppose an implementation of the trait for Unix platforms could internally store a hash map or something that associates Rust `ThreadId`s with `libc::pthread_t`s, but that introduces additional overhead... i dunno.

I have updated the code to something that will work in non-unix env's but is still rich enough to enable thread targeted thread dumps in a unix env. Let me know what you think.